Dev

valeriandr411 opened the pull request.

Commit:
Add project

Commit:
удален метод setUp

ADmitriev99 commented on pom.xml:
```diff
@@ -0,0 +1,105 @@
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.20</version>
```
Версии должны быть убраны в переменные

ADmitriev99 commented on pom.xml:
```diff
@@ -0,0 +1,105 @@
+        <dependency>
+            <groupId>com.vaadin.external.google</groupId>
+            <artifactId>android-json</artifactId>
+            <version>0.0.20131108.vaadin1</version>
```
повторение зависимостей

valeriandr411 replied:
убрала

ADmitriev99 commented on pom.xml:
```diff
@@ -0,0 +1,105 @@
+            <version>5.0.1</version>
+            <scope>compile</scope>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/org.postgresql/postgresql -->
```
таких комментариев быть не должно

valeriandr411 replied:
убрала

ADmitriev99 commented on src/main/java/check/ApiCheck.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    public static void err() throws OutOfMemoryError{
+       throw  new OutOfMemoryError("��������������� ������");
```
два десятка непонятных символов

valeriandr411 replied:
исправила, проблема с кодировкой

ADmitriev99 commented on src/main/java/check/ApiCheck.java:
```diff
@@ -0,0 +1,24 @@
+
+import static com.codeborne.selenide.Selenide.$x;
+
+public class ApiCheck {
```
что это вообще за класс?

ADmitriev99 commented on src/main/java/check/WebCheck.java:
```diff
@@ -0,0 +1,37 @@
+
+public class WebCheck {
+
+    public void test(){}
```
что это за метод?

ADmitriev99 commented on src/main/java/check/WebCheck.java:
```diff
@@ -0,0 +1,37 @@
+
+    public void test(){}
+
+    public void checkColorInCurrentCard(Color color){
```
JavaDoc здесь, до и далее

ADmitriev99 commented on src/main/java/check/WebCheck.java:
```diff
@@ -0,0 +1,37 @@
+                .should(Condition.attributeMatching("class",currentColor))
+                .getAttribute("class");
+
+        Selenide.sleep(500);
```
обосновать необходимость использования метода sleep()

ADmitriev99 commented on src/main/java/check/WebCheck.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    public void checkTextInField(String field, String text){
+        SelenideElement colorBlock= $x(".//h4[contains(text(),'Цвета')]/following-sibling::div[1]");
```
названия переменных должны быть логичны и легко считываемы

ADmitriev99 commented on src/main/java/core/PageManager.java:
```diff
@@ -0,0 +1,14 @@
+ /**
+  * Класс, который хранит текущую страницу теста
+*/
+public class PageManager {
```
лишний класс, который ничего не делает

ADmitriev99 commented on src/main/java/utils/PropertiesUtil.java:
```diff
@@ -0,0 +1,28 @@
+import java.io.InputStream;
+import java.util.Properties;
+
+public final class PropertiesUtil {
```
Этот класс нужно сделать через синглтон

ADmitriev99 commented on src/main/java/database/ConnectionManager.java:
```diff
@@ -0,0 +1,28 @@
+    private static final String USER = "db.user";
+    private static final String PASS = "db.password";
+
+    private ConnectionManager(){}
```
конструктор точно нужен?

ADmitriev99 commented on src/main/java/database/JDBCPostgreSQL.java:
```diff
@@ -0,0 +1,57 @@
+    private JDBCPostgreSQL() {
+    }
+
+    public static String getPassword(String login) {
```
дублирование кода, можно упростить

ADmitriev99 commented on src/main/java/database/JDBCPostgreSQL.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    public static String getPassword(String login) {
+        final String sql = String.format("SELECT * FROM users WHERE login = '%s'", login);
```
повторяющуюся строку в переменную

ADmitriev99 commented on src/main/java/pages/BoardsPage.java:
```diff
@@ -0,0 +1,15 @@
+
+import java.lang.reflect.Field;
+
+@Deprecated
```
зачем эта аннотация?

ADmitriev99 commented on src/main/java/pages/BoardsPage.java:
```diff
@@ -0,0 +1,15 @@
+@Deprecated
+public class BoardsPage{
+    @FindBy
+    private final String addBoardButton = "//div[contains(@class,'board-tile')]";
```
Три строки можно расположить в любом классе или в enum, отдельный создавать не надо

ADmitriev99 commented on src/main/java/pages/LogInPage.java:
```diff
@@ -0,0 +1,49 @@
+    public void googleButtonClick() {
+        $x(googleButton).click();
+    }
+    // Кнопка войти
```
комментариев быть не должно

ADmitriev99 commented on src/main/java/pages/LogInPage.java:
```diff
@@ -0,0 +1,49 @@
+    private final String googleButton = ".//a[@id='googleButton']";
+
+
+    public void typeEmail(String email) {
```
подумать над возможностью убрать дублирование

ADmitriev99 commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,17 @@
+db.password=password
+
+# loginData---------------------------------------------------------
+base.url=
```
а где?

ADmitriev99 commented on src/main/java/steps/ApiSteps.java:
```diff
@@ -0,0 +1,348 @@
+    private static final String KEY = JDBCPostgreSQL.getKey(PropertiesUtil.get("test.user"));
+    private static final String TOKEN = JDBCPostgreSQL.getToken(PropertiesUtil.get("test.user"));
+
+    //--------------------------------�����----------------------------------------//
```
это что-то на французском? к сожалению, я изучаю немецкий

ADmitriev99 commented on src/main/java/steps/ApiSteps.java:
```diff
@@ -0,0 +1,348 @@
+    public String createBoard(String name) throws JSONException {
+        JSONObject requestBody = new JSONObject()
+                .put("name", name);
+        Response response = given()
```
вынести в отдельный метод, убирать дублирование кода

ADmitriev99 commented on src/main/java/steps/ApiSteps.java:
```diff
@@ -0,0 +1,348 @@
+                .queryParam("token", TOKEN)
+                .contentType(ContentType.JSON)
+                .body(requestBody.toString())
+
```
пустая строчка?

ADmitriev99 commented on src/main/java/steps/ApiSteps.java:
```diff
@@ -0,0 +1,348 @@
+                .then()
+                .statusCode(200)
+                .extract().response();
+        response.prettyPrint();
```
Зачем вывод? Должен быть написан логгер (здесь, до и после)

ADmitriev99 commented on src/main/java/steps/ApiSteps.java:
```diff
@@ -0,0 +1,348 @@
+    }
+
+    /**
+     * �����, ����������� �������� �������� � ��������
```
Все перевести на литературный английский

ADmitriev99 commented on src/main/java/steps/WebSteps.java:
```diff
@@ -0,0 +1,103 @@
+        Selenide.open(url);
+        Selenide.sleep(500);
+    }
+    @Step("Нажать на элемент, содержащий текст '{text}' ")
```
почему здесь есть @Step, а классом выше ее нет?

ADmitriev99 commented on src/test/java/RestTest.java:
```diff
@@ -0,0 +1,80 @@
+public class RestTest {
+    private static ApiSteps apiSteps = new ApiSteps();
+
+//    @Test
```
Зачем все эти комментарии?

ADmitriev99 commented on src/test/java/RestTest.java:
```diff
@@ -0,0 +1,80 @@
+//    @Description("1. Тестирование API")
+    public void apiTest() throws JSONException {
+//Шаг 1. Создать доску "KanbanTool"
+         String idBoard = apiSteps.createBoard("Новая доска для теста");
```
каждое действие выносится в отдельный метод и обозначается аннотацией @Step

ADmitriev99 commented on src/test/java/RestTest.java:
```diff
@@ -0,0 +1,80 @@
+        apiSteps.changeCheckItemState(idCard,  idCheckItem_2,"complete" );
+        //
+
+//Шаг 13. Поставить в карточке эмоджи Палец вверх
```
а где сам шаг?

ADmitriev99 commented on src/test/java/RestTest.java:
```diff
@@ -0,0 +1,80 @@
+    public void test() throws JSONException {
+       // apiSteps.deleteBoard("63c009e1012787032ad200ac");
+       // apiSteps.getBoards();
+        String idListDane = "63c180dfa5dd3f0220c9769a";
```
это что за Hard-Code-data?

ADmitriev99 commented on src/test/java/RestTest.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+  //  @AfterClass
+    public void after() throws JSONException {
```
after dark?

ADmitriev99 commented on src/test/java/WebTest.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    public void login() {
+
```
Ctrl+Alt+L

ADmitriev99 commented on src/test/java/WebTest.java:
```diff
@@ -0,0 +1,82 @@
+
+        webSteps.clickButtonWithText("Log in");
+        final String login = PropertiesUtil.get("test.user");
+        webSteps.inputTextInField("������� ����� ����������� �����", login);
```
вынести тестовые данные в переменные

ADmitriev99 commented on src/test/java/WebTest.java:
```diff
@@ -0,0 +1,82 @@
+        login();
+        webSteps.clickElementContainsText("�������");
+
+//��� 1. ��������������, ��� �������� ��������� � ������� Done
```
ниже этой строчки проверок не проводилось

ADmitriev99 reviewed (changes requested):
Переделать.

Commit:
pom.xml: версии перенесены в properties, удалены лишние dependency
WebCheck: добавлены методы для проверки колонки и карточки в колонки
WebTest: добавлена проверка карточки в колонке

Commit:
правки

Commit:
правки

Commit:
Color: изменены коды цветов

Commit:
PageManager: добавлены методы для взаимодействия с блоками
pom:добавлена зависимость log4j
WebSteps:добавлен логер,добавлены методы для клика по заголовку, для иметации клика по клавишам, для выбора фона
application.properties: log4j

Commit:
правки

Commit:
правки

Commit:
правки

Commit:
Добавление и настройка логгера, небольшие правки

Commit:
JDBCPostgreSQL:добавлены описания к методам, вынесена константа
log4j.properties:правки в настройках вывода (теперь для всего пакета)
WebCheck: добавлен логгер
